File: app/src/main/java/com/fuckcoolapk/module/DisableUmeng.java
package com.fuckcoolapk.module;

import android.content.Context;

import com.fuckcoolapk.utils.CoolapkContext;
import com.fuckcoolapk.utils.Log;
import com.fuckcoolapk.utils.OwnSP;

import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedHelpers;

public class DisableUmeng {
    public void init() {
        if (OwnSP.INSTANCE.getOwnSP().getBoolean("disableUmeng", false)) {
            try {
                XposedHelpers.findAndHookMethod("com.umeng.commonsdk.UMConfigure", CoolapkContext.classLoader, "init", Context.class, String.class, String.class, int.class, String.class, new XC_MethodReplacement() {
                    @Override
                    protected Object replaceHookedMethod(MethodHookParam param) throws Throwable {
                        return null;
                    }
                });
            } catch (Throwable e) {
                Log.e(e);
            }
        }
    }
}

File: app/src/main/java/com/fuckcoolapk/module/DisableUmeng.kt
package com.fuckcoolapk.module

import android.content.Context
import com.fuckcoolapk.utils.CoolapkContext
import com.fuckcoolapk.utils.OwnSP
import com.fuckcoolapk.utils.ktx.hookBeforeAllMethods
import com.fuckcoolapk.utils.ktx.replaceAfterAllMethods
import com.fuckcoolapk.utils.ktx.replaceMethod
import de.robv.android.xposed.XposedHelpers

class DisableUmeng {
    fun init() {
        if (OwnSP.ownSP.getBoolean("disableUmeng", false)) {
            XposedHelpers.findClass("com.umeng.commonsdk.UMConfigure", CoolapkContext.classLoader)
                    .replaceAfterAllMethods("init") {
                        null
                    }
            XposedHelpers.findClass("com.umeng.commonsdk.UMConfigure",CoolapkContext.classLoader)
                    .replaceAfterAllMethods("preInit"){
                        null
                    }
        }
    }
}